### ISSUE TITLE
html translate with height + width= 100%: run swf in nativ resolution but stretch to 100%

### ISSUE THREAD

**Issue by Benman2785** [ruffle-rs/ruffle#13460]
some games like "The Last Stand 1" really pushing the limits of ruffle.

1. when the game renders in native resolution it is more performant
2. we can use html translate to increase output dimensions without much impact
3. so we use native resolution (like 800x420) and translate that to 100%

usage in html:
<embed src="game.swf" width=100%; height=100%; scale="translate";>

this scales/translates native resolution of swf to max-out 100% width + height of html

so in short: native resolution translated into higher output dimension

**Comment by Benman2785** [comment]
https://hoffmeister.li/flash-games/The_last_Stand.html 
uses `width/height=100%` which results in ruffle to increase render resolution and having less performance as when doing:

`
.trans {
	-webkit-backface-visibility: hidden; 
	-ms-transform: translateZ(0); /* IE 9 */
	-webkit-transform: translateZ(0); /* Chrome, Safari, Opera */
	transform: translateZ(0) scale(2.3875);
	transform-origin: 50% 50%;
	allowfullscreen="";
}
</style>
` 
`<embed src="swf/the_last_Stand_mod.swf" width="800"; height="420"; scale="exactfit"; class="trans";>`

this results in higher performance and ruffle renders in native resolution of 800x420 and the browser upscales to 2.3875 * 800x420 = 1910x1002 output resolution

// 
so with `<embed src="game.swf" width=100%; height=100%; scale="translate";>` i would want ruffle to use 800x420 as render-resolution and upscale output via browser to 100% height/width (same aspect as native)

**Comment by Benman2785** [comment]
for 3D there could be a `scale="translate3d"`

that does:
`.trans {`
`	-webkit-backface-visibility: hidden; `
`	-ms-transform: translateZ(0); /* IE 9 */`
`	-ms-transform-style: preserve-3d;`
`	-webkit-transform: translateZ(0); /* Chrome, Safari, Opera */`
`	-webkit-transform-style: preserve-3d;`
`	transform: translateZ(0) scale(2.1023);`
`	transform-style: preserve-3d;`
`	transform-origin: 50% 50%;`
`	allowfullscreen="";`
`}`

**Comment by Benman2785** [comment]
`transform: translateZ(0) scale(X.YZ);`

X.YZ should be calculated by ruffle to upscale to 100vh

**Comment by shadow-absorber** [comment]
this sounds like a good idea to shadow.... would probably help for 4k/hdpi monitors too

**Comment by Benman2785** [comment]
did someone looked into it?
actually it should only apply to web-selfhosted and the browser-addons
